tests/mesh/mesh_input.C

```diff
@@ -604,8 +604,12 @@ public:
     unsigned int n_var =
       sys.add_variable("n", Order(order), RATIONAL_BERNSTEIN);
 
-    es.init();
     dyna.add_spline_constraints(sys.get_dof_map(), sys.number(), n_var);
+    // `add_spline_constraints` secretly changed `send_list` and `global_to_local_map`
+    // by calling `dof_map.prepare_send_list()`
+    // If we initialize system earlier, then PetscVector will be out of sync
+    // We should init system only when everything is ready.
+    es.init();
```

Comment:
The current fix does work either.

Reply:
Ironically, I think one of my other open PRs would end up fixing this issue *by accident*.

But I agree with you that there's a deeper issue that needs to be fixed; if at the moment we have a test exposing the problem that's a good thing, not a bad thing.

Reply:
I am curious how you did you fix this test? Could you point me the PR?

Reply:
Check out #2808 - we basically don't wait for application-level code to apply IGA constraints any more, we make the spline node constraint connections storable in the Mesh, so the DofMap itself can translate them into dof_constraints (whether raw or by merging them with overlapping Dirichlet constraints) as soon as it needs to, and so we don't end up initializing things with one set of constraints only to have to change those constraints right afterward.